### ISSUE TITLE
NBN exports not available

### ISSUE THREAD

**Issue by kitenetter** [BiologicalRecordsCentre/iRecord#1235]
I've just gone to the "exports" folder on the server (\\192.171.199.203) where we store the files generated to send to the NBN Atlas, and none of the files have updated since 9 November.

I'm also unable to access the export admin page:
https://irecord.org.uk/exports

I'm assuming that this is a casualty of the move to Drupal 9?

The deadline for this month's upload to NBN is today, so I don't expect we will be able to send anything this month. Can we make sure this is working again by the end of February for the next round of updates.

**Comment by kitenetter** [comment]
Actually NBN say they can wait a few days for us to sort this out. @BirenRathod can you let me know if this will be easy to fix?

**Comment by BirenRathod** [comment]
@BirenRathod I was looking on old website and there is a view created by @johnvanbreda and it is not available on new irecord website. So I'm creating that but the problem is views is not identifying fields from indicia_auto_export content. Not sure at present, how long will it take?

**Comment by BirenRathod** [comment]
@kitenetter all export forms created previous like below example, are they important to keep or ok to delete?

**Comment by BirenRathod** [comment]
@kitenetter Here it is -> https://irecord.org.uk/node/8211 Like this you will fine all in below link.
https://irecord.org.uk/admin/content?title=&type=indicia_auto_export&status=All&langcode=All&page=0

**Comment by kitenetter** [comment]
Hi @BirenRathod I don't know what those individual pages do, all I normally see is the overall admin page that lists all the individual exports. I think we will need advice from @johnvanbreda

**Comment by BirenRathod** [comment]
@kitenetter Those are the same list will appear on admin once I generate the report. I just want from you to understand whether you need those or shall I just delete them.

**Comment by kitenetter** [comment]
I don't know.

**Comment by johnvanbreda** [comment]
It looks like the indicia auto export functionality hasn't been migrated to Drupal 9, but I have a vague recollection that we were going to continue using the old server/website to do this for now, rather than migrate it, as the functionality will be replaced by the Elasticsearch version anyway.
@BirenRathod is it possible to use the old server copy of iRecord to do this? If not we will have to either test the new Elasticsearch export further and add the missing configurations, or migrate the code to Drupal 9, but I don't have a lot of time spare to do the latter.

**Comment by BirenRathod** [comment]
@johnvanbreda All fields have already been migrated to new site but for some reason views is not allowing me to add fields from indicia_auto_export. you see view I crated on new site -> https://irecord.org.uk/exports

I'm creating new content type using those fields, so same could be possible to produce.

**Comment by BirenRathod** [comment]
@kitenetter the export page has now created.

**Comment by kitenetter** [comment]
Thanks @BirenRathod. I've triggered an export as a test and will let you know if it comes through okay.

**Comment by johnvanbreda** [comment]
Presumably this won't work will it? I didn't think the Indicia auto export module had been migrated.

**Comment by kitenetter** [comment]
It does not appear to have worked.

**Comment by kitenetter** [comment]
@johnvanbreda the old admin page is accessible, and says that files have continued to be created up to 17 January, but there are no new zip files on the server since 9 November 2021. Are we likely to be able to revivie this export route to keep us going until the new process is available?

**Comment by johnvanbreda** [comment]
@BirenRathod if I trigger one of the exports on the old copy of iRecord, then manually run cron, no files are generated in the d:\Drupal_Backup\indicia\exports folder. There are lots of errors in the Drupal log such as the following: 
```
Notice: Undefined index: format in ctools_custom_content_type_render() (line 192 of D:\websites\irecord\sites\all\modules\ctools\plugins\content_types\custom\custom.inc).
```
Are you able to look into this? I won't have time myself in the near future.

**Comment by BirenRathod** [comment]
@johnvanbreda the error you mentioned is not relate to exports. it comes from different source. Not sure at present where does it come from because if I access any page, it generates that.

About export, it has got some problem with PHP 7.4 and generates many deprecated or not recognize functions error. So for time being I have changed PHP to 7.3

**Comment by johnvanbreda** [comment]
Thanks @BirenRathod - I think that's sorted it. The files are now being output to D:\websites\irecord\sites\default\files\private\indicia\exports on the old live web-server - please can you arrange for @kitenetter to have access to this folder.

**Comment by BirenRathod** [comment]
@johnvanbreda & @kitenetter All zip files are stored after Nov 21. The export triggers yesterday also created in that folder. 

Anyway @kitenetter, I have copied all those in d:\Drupal_Backup\indicia\exports.

**Comment by BirenRathod** [comment]
@johnvanbreda Can I add indicia auto export module on Drupal 9? I thought, it is part of iform github repository.

**Comment by kitenetter** [comment]
Thanks @BirenRathod that will enable me to provide data to NBN for this month, so the urgent requirement is now dealt with. We'll next need to do this at the end of February.

**Comment by DavidHepper** [comment]
Do these PHP and other changes affect the ordinary exports page at https://irecord.org.uk/downloads ?

**Comment by BirenRathod** [comment]
@DavidHepper None of this will impact on downloads. NBN is separate issue which relate to our server.

**Comment by BirenRathod** [comment]
> Thanks @BirenRathod that will enable me to provide data to NBN for this month, so the urgent requirement is now dealt with. We'll next need to do this at the end of February.

@kitenetter I'm waiting for @johnvanbreda to respond. Two actions needed for export from iRecord on Pantheon.
1. add indicia auto export module.
2. @johnvanbreda to provide new Elasticsearch export.

**Comment by johnvanbreda** [comment]
@BirenRathod 

> Can I add indicia auto export module on Drupal 9? I thought, it is part of iform github repository.

Yes, but the code will need to be migrated as I don't think it was ever done for the reasons given above.

Whilst the NBN Export now works, albeit using the old site copy, I would need David Roy to approve me doing this over other work.

**Comment by DavidRoy** [comment]
@BirenRathod please do what you can to make progress on this issue.  Please close this issue and raise separate issues for the remaining work

@johnvanbreda needs to prioritise other projects until April

**Comment by BirenRathod** [comment]
@johnvanbreda Well, I leave that with you. Also need a code to update on advanced folder in indicia_api.

**Comment by kitenetter** [comment]
@BirenRathod have any new zip files been generated since 17 January, and if so are you able to transfer them into d:\Drupal_Backup\indicia\exports ?

I will need to send the data to NBN Atlas tomorrow for this month's deadline, but if that isn't possible we will just have to wait for next month.

**Comment by kitenetter** [comment]
I just tried to set up a new auto export from this page:
https://irecord.org.uk/exports

Whenever I have done this before there has been an option to give the export filter that we need for each export, in a field called "Indicia filter", but I can no longer see that field when creating a new auto export, and it is no longer being displayed on any of the existing auto exports.

So it appears that I am unable to add any new auto exports at the moment.

**Comment by BirenRathod** [comment]
>have any new zip files been generated since 17 January, and if so are you able to transfer them into d:\Drupal_Backup\indicia\exports ?
> 
> I will need to send the data to NBN Atlas tomorrow for this month's deadline, but if that isn't possible we will just have to wait for next month.

@kitenetter There is no zip files since 17 January.

**Comment by BirenRathod** [comment]
@kitenetter I have copied all zip files to D:\Drupal_backup\indicia\exports now.

**Comment by BirenRathod** [comment]
> I just tried to set up a new auto export from this page: https://irecord.org.uk/exports
> 
> Whenever I have done this before there has been an option to give the export filter that we need for each export, in a field called "Indicia filter", but I can no longer see that field when creating a new auto export, and it is no longer being displayed on any of the existing auto exports.
> 
> So it appears that I am unable to add any new auto exports at the moment.

@kitenetter I have made available the indicia filter field which gets value in old exports but the auto complete filters is not working at present. I need to understand bit more about it. like how was working in the past? is it just relate to those titles of existed export created? or is it doing something more?

**Comment by kitenetter** [comment]
Hi @BirenRathod I have another NBN upload deadline, which is tomorrow (Tuesday 5th) - these deadlines catch me surprise every time!

Can you do the same as you did last month to update the zip files into D:\Drupal_backup\indicia\exports

**Comment by BirenRathod** [comment]
@kitenetter I have copied all files in D:\Drupal_backup\indicia\exports.

**Comment by kitenetter** [comment]
Thanks @BirenRathod

**Comment by kitenetter** [comment]
Hi @BirenRathod please can you copy over the files again. I think I was supposed to send them to NBN today, but I'm hoping they'll let me do so next week!

**Comment by BirenRathod** [comment]
@kitenetter All files have now copied.

**Comment by kitenetter** [comment]
Thanks @BirenRathod

**Comment by kitenetter** [comment]
@BirenRathod there are four datasets that we normally include that are missing from D:\Drupal_backup\indicia\exports. Can you check if these are available please:
- Caddisfly (Trichoptera) records from Britain and Ireland, via iRecord
- Coleoptera records from iRecord for families not covered by a recording scheme
- Natural England iRecord Surveys
- Sheffield and Rotherham Wildlife Trust - Records

**Comment by kitenetter** [comment]
@BirenRathod please can you copy over any zip files created after 3 May.

**Comment by BirenRathod** [comment]
@kitenetter I have copied new files created after 4th  May and deleted old files from export folder now.
